src/main/java/org/springframework/data/elasticsearch/core/mapping/CreateIndexSettings.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.elasticsearch.core.mapping;
+
+import org.springframework.data.elasticsearch.core.document.Document;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Encapsulating index mapping fields, settings, and index alias(es).
+ *
+ * @author Youssef Aouichaoui
+ * @since 5.3
+ */
+public class CreateIndexSettings {
+    private final IndexCoordinates indexCoordinates;
+    private final Set<Alias> aliases;
+
+    private final Map<String, Object> settings;
+
+    @Nullable
+    private final Document mapping;
+
+    private CreateIndexSettings(Builder builder) {
+        this.indexCoordinates = builder.indexCoordinates;
+        this.aliases = builder.aliases;
+
+        this.settings = builder.settings;
+        this.mapping = builder.mapping;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public IndexCoordinates getIndexCoordinates() {
+        return indexCoordinates;
+    }
+
+    public Alias[] getAliases() {
+        return aliases.toArray(Alias[]::new);
+    }
+
+    public Map<String, Object> getSettings() {
+        return settings;
+    }
+
+    @Nullable
+    public Document getMapping() {
+        return mapping;
+    }
+
+    public static class Builder {
+        private IndexCoordinates indexCoordinates;
+        private Set<Alias> aliases = new HashSet<>();
+
+        private Map<String, Object> settings;
+
+        @Nullable
+        private Document mapping;
+
+        public Builder withIndexCoordinates(IndexCoordinates indexCoordinates) {
```

Comment:
the index coordinates are mandatory for the `CreateIndexSettings` class, right? I then would change this to a constructor `Builder(IndexCoordinates indexCoordinates)` so that it's not possible to create a builder without it.